frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -324,13 +324,27 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
 
   validateHasConditionCodeDefault(conditions: ExperimentCondition[]) {
     let defaultKeyword = this.translate.instant('home.new-experiment.design.condition.invalid.text');
-    let defaultConditionCodeErrorText = this.translate.instant('home.new-experiment.design.condition-name-validation.text')
+    let defaultConditionCodeErrorText = this.translate.instant('home.new-experiment.design.condition-name-validation.text');
     if (conditions.length >= 1 ) {
       const hasDefaultConditionCode = conditions.filter(
         condition => condition.conditionCode.toUpperCase() === defaultKeyword
       );
       if (!!hasDefaultConditionCode.length) {
-        this.conditionCodeError = defaultConditionCodeErrorText
+        this.conditionCodeError = defaultConditionCodeErrorText;
+      } else {
+        this.conditionCodeError = null;
+      }
+    }
+  }
+
+  validateHasAssignmentWeightsNegative(conditions: ExperimentCondition[]) {
+    let negativeAssignmentWeightErrorText = this.translate.instant('home.new-experiment.design.assignment-weight-negative.text');
+    if (conditions.length >= 1 ) {
+      const hasNegativeAssignmentWeights = conditions.filter(
+        condition => condition.assignmentWeight < 0
+      );
+      if (!!hasNegativeAssignmentWeights.length) {
```

Comment:
Shouldn't just `if (hasNegativeAssignmentWeights.length) {` work for displaying the error?

Reply:
Let me check, I guess in a similar such condition validation you suggested me to use !! as strict type check for undefined.

Reply:
done, without `!!` also works fine